internal/cmd/api-key/command_delete.go

```diff
@@ -32,24 +31,19 @@ func (c *command) delete(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	errs := &multierror.Error{ErrorFormat: errors.CustomMultierrorList}
-	var deleted []string
-	for _, id := range args {
+	deleted, err := deletion.DeleteResources(args, func(id string) error {
 		if r, err := c.V2Client.DeleteApiKey(id); err != nil {
-			errs = multierror.Append(errs, errors.CatchApiKeyForbiddenAccessError(err, deleteOperation, r))
-		} else {
-			deleted = append(deleted, id)
-			if err := c.deletePostProcess(id); err != nil {
-				errs = multierror.Append(errs, err)
-			}
+			return errors.CatchApiKeyForbiddenAccessError(err, deleteOperation, r)
 		}
-	}
+		return nil
+	}, func(id string) error {
```

Comment:
Can we define these functions outside of the function call? This syntax is messy, IMO.